src/test/components/AppLocalizationProvider.test.js

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import * as React from 'react';
+import { render } from '@testing-library/react';
+import { AppLocalizationProvider } from '../../components/app/AppLocalizationProvider';
+import { blankStore } from '../fixtures/stores';
+import { Provider } from 'react-redux';
+import { requestL10n, receiveL10n } from '../../actions/l10n';
+import { getIsL10nFetching, getLocalization } from '../../selectors/app';
+import { lazilyParsedBundles } from 'firefox-profiler/app-logic/l10n-bundling';
+import { ReactLocalization } from '@fluent/react';
+
+describe('AppLocalizationProvider', () => {
+  beforeEach(function() {
+    const response = (({
+      ok: true,
+      status: 200,
+      headers: {
+        get: () => 'text/plain',
+      },
+      text: () => Promise.resolve('resolved'),
+    }: any): Response);
+    window.fetch = jest.fn().mockResolvedValue(response);
+  });
+
+  afterEach(function() {
+    delete window.fetch;
+  });
+
+  function flushPromises() {
+    return new Promise(resolve => setImmediate(resolve));
+  }
+
+  it('changes the state accordingly when the actions are dispatched', () => {
+    const store = blankStore();
+    const { dispatch, getState } = store;
+
+    expect(getIsL10nFetching(getState())).toBeFalsy();
+    expect(getLocalization(getState())).toEqual(new ReactLocalization([]));
+    dispatch(requestL10n());
+    expect(getIsL10nFetching(getState())).toBeTruthy();
+    const resource = [
+      'locale',
+      'data-testid = conten-test-ApplocalizationProvider',
+    ];
+    const bundles = lazilyParsedBundles([resource]);
+    const testLocalization = new ReactLocalization(bundles);
+    dispatch(receiveL10n(testLocalization));
+    expect(getIsL10nFetching(getState())).toBeFalsy();
+    expect(getLocalization(getState())).toEqual(testLocalization);
+  });
+
+  it('AppLocalizationProvider renders its children', async () => {
+    const store = blankStore();
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <AppLocalizationProvider>
+          <div data-testid="content-test-AppLocalizationProvider" />
+        </AppLocalizationProvider>
+      </Provider>
+    );
+    await flushPromises();
+    expect(getByTestId('content-test-AppLocalizationProvider')).toBeTruthy();
```

Comment:
Suggestion:

can we use `expect(await findByTestId('content-test-AppLocalizationProvider')).toBeTruthy();` instead of `flushPromises` ?

Also I think that when you'll rebase on latest master you may have a lint issue here (use of `toBeTruthy` instead of `toBeInDocument`, but that's resolvable with `yarn lint-fix-js`

Reply:
@julienw I actually forgot why I used `flushPromises` here, I wrote that as per @canova's suggestion for some errors I had previously. I can change it and see if everything work's okay or not. I didn't have any errors `toBeTruthy` but changed to `toBeInDocument` to avoid future errors.